internal/provider/resource_consul_cluster.go

```diff
@@ -767,3 +832,37 @@ func resourceConsulClusterImport(ctx context.Context, d *schema.ResourceData, me
 
 	return []*schema.ResourceData{d}, nil
 }
+
+// buildIPAllowlist validates there are no duplicate CIDRs and returns a consul model.
+func buildIPAllowlist(cidrs []interface{}) ([]*consulmodels.HashicorpCloudConsul20210204CidrRange, error) {
+	var IPAllowList []*consulmodels.HashicorpCloudConsul20210204CidrRange
+	if len(cidrs) == 0 {
+		return IPAllowList, nil
+	}
+
+	// Seen holds validated CIDRs.
+	seen := make(map[string]struct{}, len(cidrs))
```

Comment:
Use a hashmap to validate duplicates. See the following comment on why I went with this approach:

https://github.com/hashicorp/terraform-provider-hcp/pull/455/files#r1102997970

Alternatives discussed in that comment:
- use `schema.TypeSet` for the ip_allowlist field
- don't do validation ( backend does it)